### ISSUE TITLE
ArrayBuffer in results window of console

### ISSUE THREAD

**Issue by webtic** [neo4j/neo4j#7665]
After upgrading from 3.0.1 community to 3.0.4 a query suddenly failed while it used to work fine.

The original query:

```
MATCH (c:Company)-[r:IS_MEMBER]-(g:Group {code:'xxx'})
OPTIONAL MATCH (u:User)-[:IS_ADMIN]-(c)
RETURN  DISTINCT c, r.startdate as startdate,
    r.modified as modified,
    r.approved as approved,
    u.email as admin_email
ORDER BY c.name
```

By rewriting it to:

```
MATCH (c:Company)-[r:IS_MEMBER]-(g:Group {code:'xxx'}) 
OPTIONAL MATCH (u:User)-[:IS_ADMIN]-(c) 
WITH distinct c,r,u 
RETURN c, r.startdate as startdate, r.modified as modified, r.approved as approved, u.email as admin_email ORDER BY c.name
```

It worked again.

**Neo4j Version:** 3.04
**Operating System:** Ubuntu 14.04 LTS (Amazone instance)  
**API:** console
### Steps to reproduce

Without the data it might be hard to reproduce. Data available on request.
### Expected behavior

The same output as with the 3.0.1 version
### Actual behavior

The notificaton Neo.DatabaseError.Statement.ExecutionFailed and on the location of the data-table just the text: (ArrayBuffer(null, null, null),null) (of class scala.Tuple2)

![pasted_image_01_08_16_16_43](https://cloud.githubusercontent.com/assets/3658968/17297713/23c444be-5807-11e6-9f6d-9035c6ad2ecb.png)

**Comment by averza** [comment]
I have very similar issue with Neo4j 3.0.4, the 3.0.3 version works fine. To reproduce, start Cineasts sample database (available at https://neo4j.com/developer/example-data/) and run this query:

MATCH (movie)
WHERE movie.title =~ ".*m"
RETURN DISTINCT movie.id, movie.title, movie.releaseDate, movie.genre

**Comment by timdenney** [comment]
Same error with Neoj4 3.0.4.  Seems RETURN DISTINCT will error on null values.

**Comment by Mats-SX** [comment]
Hello @webtic, @averza and @timdenney. Thank you for reporting this.

I think that the issue may have been fixed by #7936, which was first shipped with version `3.0.6`. Is it possible for you to confirm this?

Mats

**Comment by webtic** [comment]
Can confirm, just reran the query on the same data which is upgraded to 3.0.6 in the mean time and and it happily returned the data..
